src/openklant/management/commands/migrate_to_v2.py

```diff
@@ -0,0 +1,475 @@
+            raise ValueError("Invalid subjectIdentificatie data received")
+
+        if not subject_data:
+            return
+
+        if subject_class == NatuurlijkPersoon:
+            return NatuurlijkPersoon(
+                voorletters=subject_data.get("voorletters", ""),
+                voornaam=self.voornaam,
+                achternaam=self.achternaam,
+                voorvoegsel_achternaam=self.voorvoegsel_achternaam,
+                inp_bsn=subject_data.get("inp_bsn", ""),
+            )
+        elif subject_class == NietNatuurlijkPersoon:
+            return NietNatuurlijkPersoon(
+                inn_nnp_id=subject_data.get("inn_nnp_id", ""),
+                statutaire_naam=subject_data.get("statutaire_naam", ""),
+            )
+        elif subject_class == Vestiging:
+            return Vestiging(
+                bedrijfsnaam=self.bedrijfsnaam,
+                vestigings_nummer=subject_data.get("vestigings_nummer", ""),
+            )
+
+    def _get_soort_partij(self) -> SoortPartij:
+        mapping = {
+            KlantType.natuurlijk_persoon: SoortPartij.persoon,
+            KlantType.niet_natuurlijk_persoon: SoortPartij.organisatie,
+            KlantType.vestiging: SoortPartij.organisatie,
+        }
+
+        return mapping[self.subject_type]
+
+    def to_digitaal_adres(self) -> DigitaalAdres | None:
+        if not self.emailadres:
+            return
+
+        return DigitaalAdres(adres=self.emailadres)
+
+    def to_partij(self, digitaal_adres: str | None = None) -> Partij | None:
+        try:
+            subject = self._get_subject()
+        except (TypeError, ValueError):
+            logger.exception("Unable to determine subject from subjectIdentificatie")
+            return
+
+        try:
+            soort_partij = self._get_soort_partij()
+        except KeyError:
+            logger.exception("Unable to determine soortPartij from subjectType")
+            return
+
+        data = dict(
+            indicatie_actief=True,
+            indicatie_geheimhouding=False,
+            rekeningnummers=[],
+            voorkeurs_rekeningnummer=None,
+            digitale_adressen=None,
+            voorkeurs_digitaal_adres=None,
+            soort_partij=soort_partij,
+            nummer=subject.nummer if subject else "",
+            partij_identificatie=subject.migrate() if subject else None,
+        )
+
+        if digitaal_adres:
+            data["digitale_adressen"] = [dict(uuid=digitaal_adres)]
+            data["voorkeurs_digitaal_adres"] = dict(uuid=digitaal_adres)
+
+        return Partij(**data)
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
```

Comment:
Is it necessary to pass the `params` in separately? Can it not be part of the first argument?

Reply:
We could make it part of the first argument but it would require some extra work, while this currently reuses the `_params` and uses the `params` keyword available from `requests` (and does the work for us).